### PR TITLE
Detect full references to issues and pulls in commit messages

### LINKED ISSUE #10269: Issue does not close via `Fixes: <url>`
Pushing with `Fixes: https://gitea/user/repo/issues/1` in a commit message does not close a issue while `Fixes: #1` does.

The interface correctly displays a `#1` link in place of the URL. I think URLs should be resolved first before passing the string to the issue number matcher.

### PR DESCRIPTION
**Pull request by zeripath**
Fix #10269

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Comment by silverwind** [comment]
Did not seem to work with:

``` bash
git commit --allow-empty -m 'xxx - fixes: http://localhost:3500/test/test/issues/1'
```
but this did:

``` bash
git commit --allow-empty -m 'xxx - fixes: #1'
```

File: modules/markup/mdstripper/mdstripper.go [hunk 1]
```diff
@@ -72,6 +82,50 @@ func (r *stripRenderer) processString(w io.Writer, text []byte, coalesce bool) {
...
+	}
+
+	// Note: we're not attempting to match the URL scheme (http/https)
+	host := strings.ToLower(u.Host)
+	if host != "" && host != strings.ToLower(r.localhost.Host) {
+		// Process out of band
+		r.links = append(r.links, linkStr)
+		return
+	}
+
```
Comment (6543) [inline modules/markup/mdstripper/mdstripper.go:102]: and how do you handle gitea urls under subdirs?

**Comment by zeripath** [comment]
Sorry @silverwind looks like that was an initially failed attempt. The problem was that I was looking in the wrong place - I'll leave the other fix in place though.

File: modules/markup/mdstripper/mdstripper.go (hunk 1 above)
Comment (zeripath) [reply modules/markup/mdstripper/mdstripper.go:102]: See line 104 (and in the new bit line 158)

**Review by silverwind — APPROVED** [review]
Works for me. Some kind of test for this may be good to have, but not required.

**Comment by lunny** [comment]
I think tests are necessary.

**Comment by zeripath** [comment]
done

**Review by lunny — APPROVED** [review]

**Comment by zeripath** [comment]
make lg-tm work

**Pull request merged**